### DISCUSSION TITLE
New method of DoS?

### DISCUSSION THREAD

**Discussion by Nereg** [arriven/db1000n#467] [Ideas]
CVE-2022-0742: Remote Denial of Service on Linux Kernel >=5.13

Flooding icmp6 messages of type 130 or 131 is enough to exploit a memory leak in the kernel and cause the host to go out-of-memory. The volume of traffic doesn't need to be particularly high:

https://www.openwall.com/lists/oss-security/2022/03/15/3

maybe we can integrate this ?

**Comment by arriven** [comment]
@Nereg looks good, and it should be possible to easily integrate icmp requests into packetgen

**Comment by arriven** [comment]
Potentially could be a separate job though, packetgen requires root permission and it's probably not the best in terms of usability

**Comment by arriven** [comment]
https://pkg.go.dev/golang.org/x/net/icmp#example-PacketConn-NonPrivilegedPing
I'll leave it here for future reference along with https://pkg.go.dev/github.com/google/gopacket/layers#ICMPv6

**Comment by Nereg** [comment]
Working on a POC with python's [scapy](https://scapy.readthedocs.io/en/latest/introduction.html#what-makes-scapy-so-special) tool because I can't find any POCs for this CVE

**Comment by Nereg** [comment]
Sadly I can't replicate it on my kali machine.
[Wireshark screenshot](https://prnt.sc/cQVhS8jvuSz2)
[Memory graph while the attack is running](https://prnt.sc/46tBIYOjqRGg)
Kernel version is `Linux kali 5.15.0-kali3-amd64 #1 SMP Debian 5.15.15-2kali1 (2022-01-31) x86_64 GNU/Linux
` which should be vulnerable 
I used `IPv6(dst='your address')/ICMPv6EchoRequest(type=131, data=b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')` packet in scapy

**Comment by olegrybtsov** [comment]
Верните плз процент успешных попыток атак.
